### ISSUE TITLE
Freeze when opening file with typescript-tools on Windows

### ISSUE THREAD

**Issue by TisnKu** [pmizio/typescript-tools.nvim#88]
If typescript-tools is not setup, opening a file is super fast, while with typescript-tools, opening a file can take up to several seconds.

I used [typescript repo](https://github.com/microsoft/TypeScript) for testing, you can try to open a random file.

I tested in Windows and WSL2 Ubuntu, seems only Windows has this issue.

**Comment by TisnKu** [comment]
Maybe there are some loading process can be made async?

**Comment by pmizio** [comment]
Can you check by reverting before [this](https://github.com/pmizio/typescript-tools.nvim/commit/e85bc7e661e20917615cb1b6b255ae9de4fcaf87) commit?
In here I introduced recursive traversing to find tsserver.js file, maybe on windows it is slow.

**Comment by TisnKu** [comment]
@pmizio  Thanks for the prompt response. Reverting this improves a lot. The lag of first attempt to open a file reduced from 2-5s to about 300ms (according to my guess cause I don't know how to profile it).

WSL2 lag: 100ms (guess)
With no typescript-tools: almost instant

Besides, the lag only exists for first attempt to open file, for later attempts, it is very fast just like barebone neovim.

**Comment by TisnKu** [comment]
I'm happy with the prompt improvement, while still curious if there's anything can be done to improve it. 😃

**Comment by pmizio** [comment]
Putting lag caused by probably this traversing aside, this shorter one is because plugin run `npm root -g` on start to determinate where is global npm installation dir. It is fixable for sure but it is a little bit complicated.
Lag for because of taversing I try to debug faster.

**Comment by TisnKu** [comment]
@pmizio I replaced "npm root -g" with fixed path, the lag no longer exist. Maybe save the path to cache and only run "npm root -g" again when cache path does not exist or becomes invalid?

**Comment by pmizio** [comment]
It's not bad idea! I look into it!

**Comment by pmizio** [comment]
@TisnKu can you checkout above pr. It should deal with this longer delay introduced by traversing filesystem. If this will fix, then I try to deal with delay introduced by `npm root -g` in separate pr.

**Comment by TisnKu** [comment]
@pmizio  Just verified, much faster. If combined with using fixed npm root path, almost no lag at all. Thanks for the quick fix.

**Comment by pmizio** [comment]
Nice, rest of this task I try to do until monday.

**Comment by TisnKu** [comment]
@pmizio  The plugin throws an error on buffer open.
![image](https://github.com/pmizio/typescript-tools.nvim/assets/10512841/6321215b-0ef0-4511-bdce-7f42a3194bd8)
![image](https://github.com/pmizio/typescript-tools.nvim/assets/10512841/462faa82-5485-44ec-a7c6-7916001ca125)

**Comment by pmizio** [comment]
@TisnKu as always I forgot on windows I need run command using `cmd.exe /c`, should be fixed now.

**Comment by TisnKu** [comment]
@pmizio It's perfect now. Only after the buffer is opened, the lsp will start loading. I'll close the comment. Thanks.
